nilearn/region_extraction/SimpleThresholdExtractor.py

```diff
@@ -0,0 +1,128 @@
+"""
+Voxel_ratio_extractor
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from nilearn import plotting
+from nilearn._utils import check_niimg
+from nilearn._utils.niimg_conversions import concat_niimgs
+from nilearn.image import iter_img, new_img_like
+
+
+def extract_seperate_regions(img, img_data, affine, min_size):
+    """ Extract the components which have largest
+        connections given a thresholded image of the
+        components
+
+        Parameters
+        ----------
+        img: numpy array of 3D Nifti like object
+            contains a thresholded image of the components
+
+        Returns
+        -------
+        the seperated or segmented regions extracted from
+        the input image which contains a set of connected
+        maps
+
+    """
+    all_regions_img = []
+    # n_components and n_labels before removing
+    # spurious regions
+    components, labels = label(img_data)
+    label_count = np.bincount(components.ravel())
+    label_count[0] = 0
+    for label_assign, count in enumerate(label_count):
+        if count < min_size:
+            img_data[components == label_assign] = 0
+
+    # Extract n_regions and n_labels survived after
+    # minimizing the unwanted or spurious regions
+    n_components, n_regions = label(img_data)
+    for n in range(n_regions):
+        region = (n_components == n) * img_data
+        region_img = new_img_like(img, region, affine)
+        all_regions_img.append(region_img)
+        regions_4dfile = concat_niimgs(all_regions_img)
```

Comment:
You don't need `concat_niimgs` at all.

Here is a quick and dirty equivalent of your code:

```python
    labels, n_labels = label(img_data)
    labels_size = np.bincount(components.ravel())
    labels_size = 0
    regions = []
    for label_id, label_size in enumerate(labels_size):
        if label_size < min_size:
            pass
            img_data[components == label_assign] = 0
        region = img_data.copy()
        img_data[labels != label_id] = 0
        regions.append(region[..., np.newaxis])
    regions = np.concatenate(regions, axis=3)
    regions_img = new_img_like(img, regions)
return regions_img
```